onboard/nav/searcher.cpp

```diff
@@ -19,18 +20,48 @@ void Searcher::popSearchPoint( )
     return;
 }
 
+void Searcher::correctPoints( Rover * phoebe, const rapidjson::Document& roverConfig )
+{
+    double visionDistance = ( roverConfig[ "computerVision" ][ "visionDistance" ].GetDouble() );
+    const double maxDifference = 2 * visionDistance;
+
+    for( int i = 0; i < int( mSearchPoints.size() )-1; ++i )
+    {
+        Odometry point1 = mSearchPoints.at( i );
+        Odometry point2 = mSearchPoints.at( i + 1 );
+        double distance = estimateNoneuclid( point1, point2 );
+        // find distance between two points
+        if ( distance > maxDifference )
+        {
+            // generate intermediate points and add points
+            int numPoints = int( ceil( distance / maxDifference ) - 1 );
+            cout << "inserted " << numPoints << " points" << endl;
+            double newDifference = distance / ( numPoints + 1 );
+            double bearing = calcBearing( point1, point2 );
+            cout << "bearing: " << calcBearing( point1, point2 ) << endl;
+            for ( int j = 0; j < numPoints; ++j )
+            {
+                Odometry startPoint = mSearchPoints.at( i );
+                Odometry newOdom = createOdom( startPoint, bearing, newDifference, phoebe );
+                auto iterator = mSearchPoints.begin();
+                for ( int k = 0; k <= i; ++k )
```

Comment:
This for loop can be condensed into the line above:
`auto iterator = mSearchPoints.begin() + i + 1;`
If you wanted to get real fancy you could condense lines 46-52 into:
`mSearchPoints.insert(mSearchPoints.begin() + ++i, newOdom);`
I've tested both of these but you should verify that whatever you choose to go with works as well still